backend/database.ts

```diff
@@ -0,0 +1,215 @@
+import * as pathlib from 'path'
+
+import sbp from "@sbp/sbp"
+import { notFound } from 'pogo/lib/bang.ts'
+
+import '~/shared/domains/chelonia/db.js'
+import { strToB64 } from '~/shared/functions.js'
+
+const CI = Deno.env.get('CI')
+const GI_VERSION = Deno.env.get('GI_VERSION')
+const NODE_ENV = Deno.env.get('NODE_ENV')
+
+const { AlreadyExists, BadRequest, NotFound } = Deno.errors
+const dataFolder = pathlib.resolve('./data')
+const production = NODE_ENV === 'production'
+const readFileAsync = Deno.readFile
+const readTextFileAsync = Deno.readTextFile
+const writeFileAsync = Deno.writeFile
+const writeTextFileAsync = Deno.writeTextFile
+
+const dirExists = async (pathname) => {
+  try {
+    const stats = await Deno.stat(pathname)
+    return stats ?? stats.isDirectory()
+  } catch {
+    return false
+  }
+}
+
+const fileExists = async (pathname) => {
+  try {
+    const stats = await Deno.stat(pathname)
+    return stats ?? stats.isFile()
+  } catch {
+    return false
+  }
+}
+
+if (!(await dirExists(dataFolder))) {
+  await Deno.mkdir(dataFolder, { mode: 0o750, recursive: true })
+}
+
+export default (sbp('sbp/selectors/register', {
+  'backend/db/streamEntriesSince': async function (contractID: string, hash: string) {
+    let currentHEAD = await sbp('chelonia/db/latestHash', contractID)
+    if (!currentHEAD) {
+      throw notFound(`contractID ${contractID} doesn't exist!`)
+    }
+    const chunks = ['[']
+    try {
+      while (true) {
+        const entry = await sbp('chelonia/db/getEntry', currentHEAD)
+        if (!entry) {
+          console.error(`read(): entry ${currentHEAD} no longer exists.`)
+          chunks[chunks.length] = ']'
+          break
+        }
+        if (chunks.length > 1) chunks[chunks.length] = ','
+        chunks[chunks.length] = `"${strToB64(entry.serialize())}"`
+        if (currentHEAD === hash) {
+          chunks[chunks.length] = ']'
```

Comment:
Instead of this `chunks[chunks.length]` stuff, could you please preserve the previous behavior as much as possible? E.g. use `chunks.push` and keep the code identical to the previous code, except where changes are required?

Reply:
The reason I'm making this request is because every single deviation from the original source has to be re-reviewed. The fewer such deviations there are, the better. The only exception to this is boy-scouting - fixing up other people's code / making it cleaner and simpler. I don't think this change counts.

Reply:
Agree with using chunks.push() - however I had to research and use a new API as the current one didn't work in Deno.